dowhy/causal_refuters/non_parametric_sensitivity_analyzer.py

```diff
@@ -0,0 +1,270 @@
+import logging
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy
+from econml.utilities import cross_product
+from sklearn.compose import ColumnTransformer, make_column_selector
+from sklearn.linear_model import Lasso, LinearRegression
+from sklearn.model_selection import KFold
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+
+from dowhy.causal_refuters.partial_linear_sensitivity_analyzer import PartialLinearSensitivityAnalyzer
+from dowhy.causal_refuters.reisz import get_alpha_estimator
+from dowhy.utils.util import generate_moment_function, get_generic_regressor, get_numeric_features
+
+
+class NonParametricSensitivityAnalyzer(PartialLinearSensitivityAnalyzer):
+    """
+    Non-parametric sensitivity analysis for causal estimators.
+
+    Currently works only for single-dimensional treatment and binary treatment because it uses a plugin estimate of reisz function.
+
+    TODO: Extend to general treatments by implementing the loss-based reisz representer.

```

Comment:
Once this PR is accepted, let's add this as an issue or project to track this TODO

Reply:
Actually this feature is now implemented via the non-plugin reisz estimator. Thanks for catching this TODO, removed it now.